### PR TITLE
Document bulk behavior with multiple clients/documents

### PR DESCRIPTION
**Pull request by pquentin**
As suggested by @dliappis in https://github.com/elastic/rally-tracks/pull/378#discussion_r1097152616.

### CODE REVIEW

**Review by dliappis — COMMENTED** [review]
Thanks for this adding this clarification. Basically LGTM, left a few clarification questions.

File: docs/track.rst [hunk 1]
```diff
@@ -758,6 +758,12 @@ Properties
...
 * ``timeout`` (optional, defaults to ``1m``): Defines the `time period that Elasticsearch will wait per action <https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-query-params>`_ until it has finished processing the following operations: automatic index creation, dynamic mapping updates, waiting for active shards.
 
+With multiple ``clients``, Rally will split each document from ``corpora`` using as many splits as there are ``clients``. This ensures that the bulk index operations are efficiently parallelized but has the drawback that the ingestion is not done in the order of each document. For example, if ``clients`` is set to 2, one client will index the document starting from the beginning, while the other will index starting from the middle.
+
+Additionally, if there are multiple documents or corpora, Rally will try to index all documents in parallel in two ways:
```
[the remark this answers is not in the document]
Comment (pquentin) [reply docs/track.rst:763]: It's not the same thing. You can have multiple corpora, and multiple documents in each corpora. You can see this clearly when looking at the JSON structure of say geonames:

```json
  "corpora": [
    {
      "name": "geonames",
      "base-url": "https://rally-tracks.elastic.co/geonames",
      "documents": [
        {
          "source-file": "documents-2.json.bz2",
          "document-count": 11396503,
          "compressed-bytes": 265208777,
          "uncompressed-bytes": 3547613828
        }
      ]
    }
  ],
```

I know the elastic/logs track takes advantage of it: https://github.com/elastic/rally-tracks/blob/7df06b34daa50fe4d583d4bfbaea9336ff8cd278/elastic/logs/track.json#L394-L642.

So maybe I should update the doc you linked to as well!

File: docs/track.rst [hunk 2]
```diff
@@ -758,6 +758,12 @@ Properties
...
 * ``timeout`` (optional, defaults to ``1m``): Defines the `time period that Elasticsearch will wait per action <https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-query-params>`_ until it has finished processing the following operations: automatic index creation, dynamic mapping updates, waiting for active shards.
 
+With multiple ``clients``, Rally will split each document from ``corpora`` using as many splits as there are ``clients``. This ensures that the bulk index operations are efficiently parallelized but has the drawback that the ingestion is not done in the order of each document. For example, if ``clients`` is set to 2, one client will index the document starting from the beginning, while the other will index starting from the middle.
```
[the remark this answers is not in the document]
Comment (pquentin) [reply docs/track.rst:761]: Sure! Like this?

```suggestion
With multiple ``clients``, Rally will split each document using as many splits as there are ``clients``. This ensures that the bulk index operations are efficiently parallelized but has the drawback that the ingestion is not done in the order of each document. For example, if ``clients`` is set to 2, one client will index the document starting from the beginning, while the other will index starting from the middle.
```

File: docs/track.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dliappis) [reply docs/track.rst:763]: > You can have multiple corpora, and multiple documents in each corpora.

:facepalm: ofc I totally forgot about that. I think we are good with the docs as you have them then.

File: docs/track.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (dliappis) [reply docs/track.rst:761]: If we want to go the route I indicated, I could be less wordy and easier to understand (IMHO) if we provide a simple json example having a single corpus with several doc sources and the [above sentence](https://github.com/elastic/rally/pull/1666#discussion_r1101412549)

**Review by dliappis — APPROVED** [review]
This is LGTM from me, based on the latest discussion. Further improvements welcome, but it already is an incremental improvements from the current docs and effectively describe this hard to grasp concept.

**Pull request merged**